cmd/contour/serve.go

```diff
@@ -234,24 +234,39 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 
 	// step 4. register our resource event handler with the k8s informers,
 	// using the SyncList to keep track of what to sync later.
-	var informerSyncList k8s.InformerSyncList
+	informerSyncList := k8s.InformerSyncList{
+		Clients: clients,
+	}
 
-	informerSyncList.InformOnResources(clusterInformerFactory, dynamicHandler, k8s.DefaultResources()...)
+	apiResourceList, err := clients.ServerGroupsAndResources()
+	if err != nil {
+		log.WithField("context", "ServerGroupsAndResources").Errorf("could not get list of API Resources: %v", err)
+		return err
+	}
```

Comment:
Since we only pull the API resources once, do you think should document somewhere that Contour will need to be restarted after installing CRDs?

Reply:
Returning error here should cause Contour to crash and be restarted. But yes I agree docs in any case should be good to add as well since there possibly could be a race condition from the time you apply the CRDs and the time the API server is ready to serve them.